.golangci.yml

```diff
@@ -109,37 +144,188 @@ linters-settings:
+      - stringintconv
+      - structtag
+      - testinggoroutine
+      - tests
+      - unmarshal
+      - unreachable
+      - unsafeptr
+      - unusedresult
+      - unusedwrite
   loggercheck:
     kitlog: false
     klog: false
     require-string-key: true
     no-printf-like: true
   paralleltest:
     ignore-missing: true
+  prealloc:
+    for-loops: true
   revive:
-    enable-all-rules: true
+    # https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md
     rules:
       - name: add-constant
         disabled: true
+      - name: argument-limit
+        arguments: [6]
+      - name: atomic
+      - name: banned-characters
+        arguments: ["Ω", "Σ", "σ", "7"]
+      - name: bare-return
+      - name: blank-imports
+      - name: bool-literal-in-expr
+      - name: call-to-gc
       - name: cognitive-complexity
         disabled: true
+      - name: comment-spacings
+      - name: comments-density
+        disabled: true
+      - name: confusing-naming
+      - name: confusing-results
+      - name: constant-logical-expr
+      - name: context-as-argument
+        arguments:
+          - allowTypesBefore: "*testing.T"
+      - name: context-keys-type
       - name: cyclomatic
         disabled: true
+      - name: datarace
+      - name: deep-exit
+      - name: defer
+        arguments:
+          - ["call-chain", "loop", "method-call", "recover", "immediate-recover", "return"]
       - name: dot-imports
-        disabled: true
+        arguments:
+          - { allowedPackages: ["github.com/onsi/ginkgo/v2","github.com/onsi/gomega"] }
+      - name: duplicated-imports
+      - name: early-return
+      - name: empty-block
+      - name: empty-lines
+      - name: enforce-map-style
+        arguments: ["make"]
+        exclude: ["TEST"]
+      - name: enforce-repeated-arg-type-style
+        arguments:
+          - { funcArgStyle = "any", funcRetValStyle = "short" }
+      - name: enforce-slice-style
+      - name: error-naming
+      - name: error-return
+      - name: error-strings
+      - name: errorf
       - name: exported
         disabled: true
+        arguments:
+          - "checkPrivateReceivers"
+          - "sayRepetitiveInsteadOfStutters"
+      - name: file-header
+        disabled: true
+      - name: flag-parameter
       - name: function-length
         disabled: true
-      - name: max-public-structs
-        disabled: true # the api/* pkgs have lots of structs
+      - name: function-result-limit
+        arguments: [3]
+      - name: get-return
+      - name: identical-branches
+      - name: if-return
+      - name: import-alias-naming
+        arguments:
+          - "^[a-z][a-z0-9]{0,}$"
+      - name: import-shadowing
+      - name: imports-blocklist
+        disabled: true
+      - name: increment-decrement
+      - name: indent-error-flow
       - name: line-length-limit
         disabled: true
+      - name: max-control-nesting
+        arguments: [3]
+      - name: max-public-structs
```

Comment:
Also suggest to disable that, some public APIs will fail that linter

Reply:
public APIs are removed from this check